### PR TITLE
Touch pin definition overhaul

### PR DESCRIPTION
**Pull request by bugadani**
This PR merges the different touch macros, which also enables implementing the Touch trait directly - uncovering two missing pins.

### CODE REVIEW

**Review by bjoernQ — APPROVED** [review]
Nice

**Review by JurajSadel — APPROVED** [review]
Thanks!

File: esp-hal/src/gpio/mod.rs [hunk 1]
```diff
@@ -177,15 +177,6 @@ pub enum Pull {
     Down,
 }
 
-/// RTC input pin mode
```
Comment (MabezDev) [inline esp-hal/src/gpio/mod.rs:180]: We're these unused before?

Comment (bugadani) [reply esp-hal/src/gpio/mod.rs:180]: apparently not

**Pull request merged**